ansible_base/rbac/models/role.py

```
@@ -263,6 +263,195 @@ def give_permission(self, actor, content_object):
+            team_ids = {actor.pk for _, actor, _, _, _ in all_triples[len(user_permissions) :]}
+            RoleTeamAssignment.objects.filter(object_role_id__in=all_or_ids, team_id__in=team_ids).delete()
+
+        # -- orphan cleanup --
+        orphaned = ObjectRole.objects.filter(id__in=all_or_ids, users__isnull=True, teams__isnull=True)
+        orphaned_ids = set(orphaned.values_list('id', flat=True))
+        surviving = {or_ for or_ in or_lookup.values() if or_.pk not in orphaned_ids}
+
+        # -- recomputation --
+        rd_has_team_perm = {}
+        recompute_team_ids = set()
+        for (rd_id, ct_id, oid), or_ in or_lookup.items():
```

Ruff (0.16.0)

[warning] Loop control variable `ct_id` not used within loop body

(B007)